plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreSchema.java

```diff
@@ -434,6 +434,14 @@ private void cacheTableInheritance(DBRProgressMonitor monitor) throws DBExceptio
         }
     }
 
+    private void resetPartitionsInheritance(DBRProgressMonitor monitor) throws DBException {
+        List<PostgreTable> partitions = getTableCache().getTypedObjects(monitor, this, PostgreTable.class).stream()
+            .filter(PostgreTableBase::isPartition).collect(Collectors.toList());
+        for (PostgreTable partition : partitions) {
+            partition.resetSuperInheritance();
+        }
```

Comment:
Maybe just use a simple `for` loop with `table.isPartition()` check? What's the point of using streams here?